### PR TITLE
Ny 1uqtg expire announcements

### PR DESCRIPTION
**Pull request by nishanty13**

### CODE REVIEW

File: mysite/announcements/tasks.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        for u in user_list:
+            print("{} {} {}".format(u, announcement.title, announcement.message))
+        announcement_id.append(announcement.id)
+
+    Announcements.objects.filter(id__in=announcement_id).update(sent_at=current_time)
+
+@task()
+def expire_announcements():
+    current_time = timezone.now()
+    expired_announcements = Announcements.objects.exclude(sent_at=None).filter(has_expired=False).filter(date_time_expire__lte=current_time)
```
Comment (tarungarg546) [inline mysite/announcements/tasks.py:37]: We can combine these both `.filter` into one using 
`.filter(has_expired=False, date_time_expire__lte=current_time)`

File: mysite/announcements/tasks.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    announcement_id = []
```
Comment (tarungarg546) [inline mysite/announcements/tasks.py:38]: `expired_announcement_ids` would be a better name IMO.

File: mysite/mysite/settings.py [hunk 3]
```diff
@@ -119,3 +122,30 @@
...
+CELERY_TIMEZONE = 'Asia/Kolkata'
+
+CELERY_BEAT_SCHEDULE = {
+    'check-scheduled-announcements': {
+        'task': 'announcements.tasks.check_scheduled_announcements',
+        'schedule': crontab('*/10'),
+    },
+    'expire-announcements': {
+        'task': 'announcements.tasks.expire_announcements',
+        'schedule': crontab(),
```
Comment (tarungarg546) [inline mysite/mysite/settings.py:149]: Again why 1 min, that's too aggressive.

File: mysite/announcements/tasks.py (hunk 1 above)
Comment (tarungarg546) [inline mysite/announcements/tasks.py:37]: Why are we excluding `sent_at=None`?

File: mysite/announcements/models.py [hunk 4]
```diff
@@ -1,24 +1,39 @@
...
+    has_expired = models.BooleanField(default=False, help_text='Marked true when an announcement expires')
+
     def __str__(self):
         return self.title
 
     def clean(self):
-        if self.date_time_expire <= timezone.now():
-            raise ValidationError("The date and time can't be in past")
\ No newline at end of file
+        if (self.date_time_to_publish <= timezone.now()) or (self.date_time_expire <= timezone.now()):
```
Comment (tarungarg546) [inline mysite/announcements/models.py:31]: Should we also have a validation that date_time_to_publish<=date_time_expire?

**Review by tarungarg546 — CHANGES_REQUESTED** [review]

File: mysite/announcements/models.py (hunk 4 above)
Comment (nishanty13) [reply mysite/announcements/models.py:31]: Yes, makes sense

File: mysite/announcements/tasks.py (hunk 1 above)
Comment (nishanty13) [reply mysite/announcements/tasks.py:37]: Those announcements are not sent yet so how can they expire

**Pull request closed without merging**